sdk/core/core/src/az_http_response.c

```diff
@@ -216,6 +219,9 @@ az_http_response_get_next_header(az_http_response* response, az_pair* out_header
     out_header->value = az_span_slice(*reader, 0, offset_value_end);
     // moving reader. It is currently after \r was found
     *reader = az_span_slice_to_end(*reader, offset);
+
+    // Remove white spaces from header name https://github.com/Azure/azure-sdk-for-c/issues/604
+    out_header->value = _az_span_trim_white_space_from_end(out_header->value);
```

Comment:
trim both sides, right?

Reply:
No need to trim left. It was already done by the logic of the http_response parser, It will read everything after the colon `:` and until a non white-space if found. So we don't need to do it again